### PR TITLE
NixOS: support dry activation

### PR DESCRIPTION
**Pull request by veehaitch**
An attempt to address #54 / #55. Unfortunately, the dependency of `agenixRoot` on the `specialfs` activation script prevented me from fixing the issue entirely. The `specialfs` activation script does not (yet) support dry activation. Ideally, we would add dry activation support for `specialfs` upstream.

Also fixed an f-string related issue with the integration test. I didn't bother to add proper dry activation support for the `installSSHHostKeys` test.

### CODE REVIEW

**Comment by ryantm** [comment]
It would be nice to make the dry-activate messages more informative:

1. decryption fails:
     > RED COLOR [agenix] dry-activate: failed to decrypt secret for path ${path}

     Ideally, it would also say why it failed to decrypt. Like is it missing the decryption key, or did it just not decrypt?
2. decryption succeeds:
     > [agenix] dry-activate: decrypted secret for path ${path}

File: modules/age.nix [hunk 1]
```diff
@@ -17,15 +17,28 @@ let
...
     (
       umask u=r,g=,o=
       LANG=${config.i18n.defaultLocale} ${ageBin} --decrypt ${identities} -o "$TMP_FILE" "${secretType.file}"
     )
     chmod ${secretType.mode} "$TMP_FILE"
     chown ${secretType.owner}:${secretType.group} "$TMP_FILE"
-    mv -f "$TMP_FILE" '${secretType.path}'
+
+    if [[ "$NIXOS_ACTION" == "dry-activate" ]]; then
+      fcho "[agenix] dry-run, not moving decrypted secret"
```
Comment (ryantm) [inline modules/age.nix:37]: fcho?

Comment (veehaitch) [reply modules/age.nix:37]: Sorry, I had to leave in a rush and wasn't able to test this properly yet. Just wanted to share the progress I already made as draft.

**Comment by veehaitch** [comment]
Tested with https://github.com/NixOS/nixpkgs/commit/b3d047de5c0a613efba8edf4785f6121dafc4f95

**Comment by veehaitch** [comment]
> It would be nice to make the dry-activate messages more informative:

While I still agree that this might be a useful addition, I'm hesitating to make the bash code any more complex. Feel free to add any commits to the branch @ryantm.

**Comment by veehaitch** [comment]
I don't think this PR is relevant anymore, apart from https://github.com/ryantm/agenix/pull/56/commits/2f1666b80264365de48018bb0535834684a8a8a8 which is still an issue, as far as I can tell.

**Pull request closed without merging**